code/app/src/main/java/com/example/bread/view/AddMoodEventActivity.java

```diff
@@ -0,0 +1,181 @@
+package com.example.bread.view;
+
+import android.Manifest;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.CheckBox;
+import android.widget.EditText;
+import android.widget.Spinner;
+import android.widget.Toast;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+import com.example.bread.R;
+import com.example.bread.model.MoodEvent;
+import com.example.bread.model.Participant;
+import com.example.bread.repository.MoodEventRepository;
+import com.example.bread.repository.ParticipantRepository;
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class AddMoodEventActivity extends AppCompatActivity {
+    private static final int LOCATION_PERMISSION_REQUEST_CODE = 1;
+    private Spinner emotionalStateSpinner, socialSituationSpinner;
+    private EditText eventTitleEditText, reasonEditText, triggerEditText;
+    private CheckBox locationCheckbox;
+    private Button saveButton;
+    private MoodEventRepository moodEventRepository;
+    private ParticipantRepository participantRepository;
+    private FusedLocationProviderClient fusedLocationClient;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_add_mood_event);
+
+        // Tie UI elements to variables
+        emotionalStateSpinner = findViewById(R.id.emotionalStateSpinner);
+        reasonEditText = findViewById(R.id.reasonEditText);
+        socialSituationSpinner = findViewById(R.id.socialSituationSpinner);
+        saveButton = findViewById(R.id.saveButton);
+        eventTitleEditText = findViewById(R.id.eventTitleEditText);  // New field
+        triggerEditText = findViewById(R.id.triggerEditText);        // New field
+        locationCheckbox = findViewById(R.id.locationCheckbox);      // New field
+
+        // Set up repositories and location client
+        moodEventRepository = new MoodEventRepository();
+        participantRepository = new ParticipantRepository();
+        fusedLocationClient = LocationServices.getFusedLocationProviderClient(this);
```

Comment:
This implementation has already been written in `LocationHandler` we don't need duplicate impl. Please use that.

If you need some implementation that is missing in the class feel free to add it there, So that everyone can use it rather than copying it everywhere.